src/reflect/scala/reflect/internal/util/SourceFile.scala

```diff
@@ -171,12 +171,31 @@ class BatchSourceFile(val file : AbstractFile, content0: Array[Char]) extends So
   }
 
   private lazy val lineIndices: Array[Int] = {
+    def countEOL(cs: Array[Char]): Int = {
+      var i, c = 0
+      while (i < cs.length){
+        if (isAtEndOfLine(i))
+          c += 1
+        i += 1
+      }
+      c
+    }
     def calculateLineIndices(cs: Array[Char]) = {
-      val buf = new ArrayBuffer[Int]
-      buf += 0
-      for (i <- 0 until cs.length) if (isAtEndOfLine(i)) buf += i + 1
-      buf += cs.length // sentinel, so that findLine below works smoother
-      buf.toArray
+      // count EOL characters in cs
+      val res = new Array[Int](countEOL(cs) + 2)
+      res(0) = 0
+      res(res.length - 1) = cs.length
+      // fill res with positions of EOL 
+      var j = 1
+      var i = 0
```

Comment:
I'm just watching the baseball game, but I think `var i, j = 0` and then switch lines 193-194, increment first.

Reply:
Not really. The `i` counter indicates where in the `cs` array we are in, and it is the variable to iterate through the loop. In Java, this would have been written `for (int i = 0; i < cs.length ; i ++)`

The `j` counter indicates which position in `res` we have to write next. Note that `res(0)` is set to `0` outside the loop, so we have to start from 1.

Reply:
That's why `j += 1; res(j) = I+1` at L193. `res(++j) = i+1`. Either way, sorry to distract you over counters.

Reply:
Ok, I understand it now. I have applied the suggestion.